util/pingpong.c

```diff
@@ -565,6 +565,18 @@ static int pp_send_name(struct ct_pingpong *ct, struct fid *endpoint)
 	if (ret < 0)
 		goto fn;
 
+	PP_DEBUG("Sending address format\n");
+	if (ct->fi) {
+		ret = pp_ctrl_send(ct, (char *) &ct->fi->addr_format,
+				   sizeof(ct->fi->addr_format));
+	} else {
+		ret = pp_ctrl_send(ct, (char *) &ct->fi_pep->addr_format,
+				   sizeof(ct->fi_pep->addr_format));
+
+	}
+	if (ret < 0)
+		return ret;
```

Comment:
You should `goto fn` instead of returning here, otherwise `local_name` will leak.

Reply:
thanks - will update